lua/packer/luarocks.lua

```diff
@@ -300,8 +327,12 @@ local function clean_packages(rocks, results, disp)
       local to_remove = {}
       for _, package in ipairs(installed_packages) do to_remove[package.name] = package end
       for _, rock in pairs(rocks) do
-        if type(rock) == 'table' then
-          if to_remove[rock[1]] and to_remove[rock[1]].version == rock[2] then
+        if vim.tbl_islist(rock) then
```

Comment:
I'm probably missing something obvious, but I'm a bit confused by the `tbl_islist` cases here and in `ensure_packages`. Shouldn't the `rocks` table contain as values only rocks specified as strings or as non-list tables (i.e. with version or server keys)? When do we encounter the list case?

Reply:
@wbthomason the it appeared that before a user could specify rocks using `use_rocks {'cjson', 'penlight', 'luaformatter}` so those are the cases that the `tbl_islist`. I might have misunderstood, if that isn't a required format since lists of tables aren't supported then I can just have it support a table or a string.

Reply:
You're right that this format is allowed, but - I think - by the time we reach `clean_packages` or `ensure_packages` - the `rocks` table had been converted to a non-list table. In particular, looking at `packer.lua:117` and `packer.lua:193`, I'm pretty sure we always perform this conversion before any call to a function in the `luarocks` module, so this case should be redundant. I could be forgetting something, though!

Reply:
@wbthomason you're right just tried without those checks and plugins are still added and removed fine and as you say things are correctly flattened. I think my thinking at the time was related to other code paths other than `use_rocks` but I guess those specify packages as strings